### PR TITLE
feat(browser-starfish): add example images in resource summary"

### PR DESCRIPTION
**Pull request by DominikB2014**
Work for #60482 

Adds sample images to the resources summary. For example:

image

### CODE REVIEW

**Comment by DominikB2014** [comment]
Gonna convert to draft as the backend changes for sorting are ready! Will add to this

File: static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    sorts: [{field: HTTP_RESPONSE_CONTENT_LENGTH, kind: 'desc'}],
+    limit: 100,
+  });
+
+  const uniqueResources = new Set();
+
+  const filteredResources = imageResources.data
+    .filter(resource => {
+      const fileName = getFileNameFromDescription(resource[SPAN_DESCRIPTION]);
+      const key = `${fileName}`;
```
Comment (AbhiPrasad) [inline static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx:27]: filename is not a string? Why do we need the cast?

File: static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+const ImageWrapper = styled('div')`
+  display: grid;
+  grid-template-columns: repeat(auto-fill, ${imageWidth});
+  gap: 30px;
+`;
+
+const getFileNameFromDescription = (description: string) => {
+  return description.split('/').pop()?.split(/[?#]/)[0] || '';
```
Comment (AbhiPrasad) [inline static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx:83]: can we parse with `new URL` instead?

File: static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx (hunk 1 above)
Comment (DominikB2014) [reply static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx:27]: Ooops, I had this key as something different before so it was needed, i'll update it 👍

File: static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx (hunk 2 above)
Comment (DominikB2014) [reply static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx:83]: 👍

File: static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+const ImageWrapper = styled('div')`
+  display: grid;
+  grid-template-columns: repeat(auto-fill, ${imageWidth});
+  gap: 30px;
+`;
+
+const getFileNameFromDescription = (description: string) => {
+  const url = new URL(description);
```
Comment (AbhiPrasad) [inline static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx:82]: this can throw if invalid URL, we should handle that case.

**Review by AbhiPrasad — APPROVED** [review]

**Comment by asottile-sentry** [comment]
reverting: broke the build

**Comment by DominikB2014** [comment]
@asottile-sentry looks like this failed when merged because it wasn't up to date with master :(

I guess this is unavoidable unless we enable "require up to date branches" or do something smarter to determine when to rerun certain parts of CI other then a conflict (like same file changed or something)?

**Pull request merged**